src/request.ts

```diff
@@ -11,9 +11,12 @@ const newRequestFromIncoming = (
   incoming: IncomingMessage | Http2ServerRequest
 ): Request => {
   const headerRecord: [string, string][] = []
-  const len = incoming.rawHeaders.length
-  for (let i = 0; i < len; i += 2) {
-    headerRecord.push([incoming.rawHeaders[i], incoming.rawHeaders[i + 1]])
+  const rawHeaders = incoming.rawHeaders
+  for (let i = 0; i < rawHeaders.length; i += 2) {
+    const {[i]: key, [i + 1]: value} = rawHeaders
+    if (key.charCodeAt(0) !== /*:*/ 0x3a) {
+      headerRecord.push([key, value])
+    }
```

Comment:
Undici had a bug that allowed ':', which was not allowed in the header name, but now that it is fixed.

Reply:
Are these lines really necessary? These affect performance, so I would like to remove them if it is not a practical problem.

Reply:
What version of undici are you using?
The whatwg fetch spec does not allow h2 pseudo-headers containing `:` and throws an error. This check is necessary.
```js
new Headers({ ":authority": "localhost" }); // error
```

Reply:
If you have this bug, `:` is still allowed.
```shell
> $ node
Welcome to Node.js v20.11.0.
Type ".help" for more information.
> new Headers().set("x", "0") instanceof Map
true
```

Reply:
I see!

@usualoma Can you see this? I think we have to go with this code, but any thoughts?

Reply:
Hi @tsctx
Thanks for creating the PR, it is very informative!

@yusukebe 
`newRequestFromIncoming()` is often not called on simple requests, and `key.charCodeAt(0) ! == /*:*/ 0x3a`, I do not think the overhead caused by such code is a problem.